### ISSUE TITLE
Accepting Invalid Type Annotation in Record projection by expression 

### ISSUE THREAD

**Issue by mageshb** [dhall-lang/dhall-haskell#1054]
Following code is accepted by dhall even though x and y are annotated with `Text` instead of expected `Natural`.

```
let r = {x = 1, y = 2, z = True} in r.({x : Text, y : Text})
```

**Comment by sjakobi** [comment]
This bug has already been fixed in https://github.com/dhall-lang/dhall-haskell/pull/1027.

**Comment by mageshb** [comment]
Thank You

**Comment by Gabriel439** [comment]
For people who run across this issue, version 1.24.0 of the interpreter has the fix for this bug

**Comment by mageshb** [comment]
I got this issue on 1.24.0 from Hackage.
```
dhall version
Haskell package version: 1.24.0
Standard version: none
```

**Comment by Gabriel439** [comment]
Oh, then this is a different issue.  I'll reopen this

**Comment by Gabriel439** [comment]
Oh wait, sorry.  This is fixed, but not in 1.24.0.  The fix has not been published yet, but will be part of release 1.25.0 (ETA: 1-2 weeks from now)

I was confusing this issue with https://github.com/dhall-lang/dhall-haskell/pull/965, which is an unrelated issue that was fixed in 1.24.0.
